tokio/src/runtime/task/join.rs

```diff
@@ -252,7 +252,41 @@ impl<T> JoinHandle<T> {
     }
 
     /// Returns a new `AbortHandle` that can be used to remotely abort this task.
-    pub(crate) fn abort_handle(&self) -> super::AbortHandle {
+    ///
+    /// Awaiting a task cancelled by the `AbortHandle` might complete as usual if the task was
+    /// already completed at the time it was cancelled, but most likely it
+    /// will fail with a [cancelled] `JoinError`.
+    ///
+    /// ```rust
+    /// use tokio::{time, task};
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///    let mut handles = Vec::new();
```

Comment:
I know the duration is long enough that this is quite unlikely to fail, but we strongly want to avoid tests that depend on timing. We can fix it by adding a (hidden) line like this:
```suggestion
    /// async fn main() {
    /// #  tokio::time::pause();
    ///    let mut handles = Vec::new();
```